components/push/src/lib.rs

```diff
@@ -4,6 +4,181 @@
 
 #![allow(unknown_lints)]
 #![warn(rust_2018_idioms)]
+//! # Rust Push Component
```

Comment:
If having these live here in the Rustdoc works out well, we could consider removing them from the README and replacing it with a more succinct overview and a link to the generated docs. That's for a future PR though.